transformations/transformations/pool_allocator.py

```diff
@@ -121,6 +122,7 @@ def __init__(self, block_dim,
         self.local_ptr_var_name_pattern = local_ptr_var_name_pattern
         self.directive = directive
         self.check_bounds = check_bounds
+        self.trafo_dict = {}
```

Comment:
Does this really need to be a property on the Transformation class object? It gets overwritten for every item/subroutine that is being processed, so shouldn't really be stored as a state on the object.